File: doc/src/codes/Users_Manual/controller_2.cs
using AUTD3Sharp.Gain;
using AUTD3Sharp.Utils;

await autd.Group(dev =>
    {
        return dev.Idx switch
        {
            0 => "null",
            1 => "focus",
            _ => null
        };
    })
    .Set("null", new Null())
    .Set("focus", new Focus(new Vector3d(x, y, z)))
    .SendAsync();

File: doc/src/codes/Users_Manual/controller_2.cpp
autd.group([](const autd3::Device& dev) -> std::optional<const char*> {
      if (dev.idx() == 0) {
        return "null";
      } else if (dev.idx() == 1) {
        return "focus";
      } else {
        return std::nullopt;
      }
    })
    .set("null", autd3::gain::Null())
    .set("focus", autd3::gain::Focus(autd3::Vector3(x, y, z)))
    .send_async()
    .get();